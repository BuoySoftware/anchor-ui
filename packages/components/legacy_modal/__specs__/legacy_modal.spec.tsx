import { Button } from "@buoysoftware/anchor-button";

import {
  LegacyModal,
  LegacyModalActions,
  LegacyModalContent,
  LegacyModalHeader,
} from "../src";
import { render } from "../test_utils";

const MockLegacyModal: React.FC = (): React.ReactElement => {
  return (
    <LegacyModal ariaHideApp={false} isOpen={true} testId="test-legacy-modal">
      <LegacyModalHeader title="Test Legacy Modal" />
      <LegacyModalContent>Test modal contents</LegacyModalContent>
      <LegacyModalActions>
        <Button onClick={jest.fn()} colorScheme="basic" mr="xxs">
          Cancel
        </Button>
        <Button onClick={jest.fn()} colorScheme="primary">
          Save
        </Button>
      </LegacyModalActions>
    </LegacyModal>
  );
};

describe("<Modal />", () => {
  it("renders correctly", () => {
    const modal = render(<MockLegacyModal />);

    expect(modal.baseElement).toMatchSnapshot();
  });
});
