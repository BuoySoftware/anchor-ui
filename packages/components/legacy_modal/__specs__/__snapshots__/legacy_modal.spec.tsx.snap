// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
.c8 {
  box-sizing: border-box;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
  margin-right: 0.25rem;
}

.c11 {
  box-sizing: border-box;
  color: inherit;
  width: s;
  height: s;
}

.c13 {
  box-sizing: border-box;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 1.25rem;
  padding-right: 1.25rem;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c10 {
  border-style: solid;
  border-width: 1px;
  background-color: #FFFFFF;
  border-color: #7F8387;
  color: #101112;
}

.c10:hover {
  background-color: #F5F6F7;
}

.c10:active {
  background-color: #D9DDE1;
}

[disabled] .c7 {
  background-color: #FFFFFF;
  border-color: #BABEC3;
  color: #9CA1A5;
}

.c14 {
  border-style: solid;
  border-width: 1px;
  background-color: #276EF1;
  border-color: transparent;
  color: #FFFFFF;
}

.c14:hover {
  background-color: #1E54B7;
}

.c14:active {
  background-color: #174291;
}

[disabled] .c7 {
  background-color: #EBEDF0;
  border-color: transparent;
  color: #9CA1A5;
}

.c12 {
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Segoe UI, Roboto,Helvetica Neue,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  border: 2px solid transparent;
  background: transparent;
}

.c6:focus-visible {
  background: rgba(255,255,255,0.01);
  border-radius: 4px;
  border-color: rgba(255,255,255,0.01);
  box-shadow: 0px 0px 0px 2px #5b91f4;
  outline: none;
}

.c0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  padding-left: 2rem;
  padding-right: 2rem;
}

.c1 {
  box-sizing: border-box;
  color: #101112;
  width: xl;
  height: xl;
}

.c3 {
  box-sizing: border-box;
  max-height: 60vh;
  overflow: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.75rem;
  padding-bottom: 1.75rem;
  padding-left: 2rem;
  padding-right: 2rem;
}

.c4 {
  box-sizing: border-box;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 1.25rem;
  margin-left: 2rem;
  margin-right: 2rem;
  padding-bottom: 1.75rem;
}

.c5 {
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Segoe UI, Roboto,Helvetica Neue,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 1.5rem;
  font-weight: 600;
  line-height: 1.75rem;
}

<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div>
    <div
      id="test-root"
      style="position: relative;"
    />
  </div>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay"
      style="position: fixed; top: 0px; left: 0px; right: 0px; bottom: 0px; background-color: rgba(10, 25, 53, 0.64); display: flex; align-items: flex-start; justify-content: center; z-index: 99;"
    >
      <div
        aria-modal="true"
        class="ReactModal__Content"
        data-testid="test-legacy-modal"
        role="dialog"
        style="background: rgb(255, 255, 255); overflow: auto; border-radius: 30px; outline: none; padding: 0px; background-color: rgb(255, 255, 255); margin-top: 4.25rem; width: 527px;"
        tabindex="-1"
      >
        <div
          class="c0 "
          display="flex"
        >
          <div
            class="c1 c2"
            color="text.primary"
            data-testid=""
            font-size="font-size-400"
            font-weight="600"
          >
            Test Legacy Modal
          </div>
        </div>
        <div
          class="c3 "
          display="flex"
          overflow="auto"
        >
          Test modal contents
        </div>
        <div
          class="c4 c5 "
          display="flex"
        >
          <button
            class="c6"
            type="button"
          >
            <div
              class="c7 c8 c9 c10"
              display="flex"
              height="40"
            >
              <span
                class="c11 c12"
                color="inherit"
                font-size="font-size-100"
                font-weight="600"
                text-decoration="none"
              >
                Cancel
              </span>
            </div>
          </button>
          <button
            class="c6"
            type="button"
          >
            <div
              class="c7 c13 c9 c14"
              display="flex"
              height="40"
            >
              <span
                class="c11 c12"
                color="inherit"
                font-size="font-size-100"
                font-weight="600"
                text-decoration="none"
              >
                Save
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
